### PR TITLE
MGDAPI-4618 - address all linter issues with golangci-lint

### PR DESCRIPTION
**Pull request by kramaranya**
# Issue link

[Linter Epic](https://issues.redhat.com/browse/MGDAPI-4593)

# What

Address all the errors from the Linter Epic: staticcheck and gosimple

# Verification steps

- Checkout this PR
- Run `make test/lint`
- Confirm there are no errors in the output
- Code review

### CODE REVIEW

**Review by Boomatang — CHANGES_REQUESTED** [review]
Couple of small points.

File: pkg/products/threescale/reconciler_test.go
Comment (kramaranya) [reply pkg/products/threescale/reconciler_test.go]: Updated

**Comment by Fiona-Waters** [comment]
Followed verification steps and can confirm that there were no errors in the output. Code changes look good. Will leave it to  @Boomatang to lgtm if he is happy with newest changes.

**Review by Boomatang — APPROVED** [review]
Changes look good to me
/lgtm

**Pull request merged**
